server/src/compression.rs

```diff
@@ -0,0 +1,467 @@
+/* Copyright 2022 The ModelarDB Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+//! Compress `UncompressedSegments` provided by [`StorageEngine`] using the
```

Comment:
I cannot link to `UncompressedSegment` as `segment` is defined as `mod segment` instead of `pub mod segment` in `storage.rs`. and I did not want to change `storage.rs` in this PR.

Reply:
That is fine, we can change it at a later point if necessary.